### PR TITLE
es6ified js

### PR DESCRIPTION
**Pull request by plexigras**
#3216

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3217)

### CODE REVIEW

File: qutebrowser/javascript/.eslintrc.yaml [hunk 1]
```diff
@@ -45,3 +43,15 @@ rules:
...
+     no-implicit-globals: "off"
+     no-bitwise: "off"
+     strict: "off"
+     no-undef: "off"
+     no-param-reassign: "off"
+     no-eq-null: "off"
+     no-ternary: "off"
+     curly: "off"
+     eqeqeq: "off"
+     max-statements: "off"
```
Comment (The-Compiler) [inline qutebrowser/javascript/.eslintrc.yaml]: Why turn all those off? Most of them seem useful to me.

File: qutebrowser/javascript/pac_utils.js [hunk 2]
```diff
@@ -48,201 +48,196 @@ function dnsDomainIs(host, domain) {
...
+    "NOV": 10,
+    "DEC": 11,
+};
+
+function weekdayRange(...args) {
     function getDay(weekday) {
-        if (weekday in wdays) {
-            return wdays[weekday];
-        }
+        if (weekday in wdays) return wdays[weekday];
```
Comment (The-Compiler) [inline qutebrowser/javascript/pac_utils.js]: I'd rather have `return` statements on their own lines.

File: qutebrowser/javascript/history.js [hunk 3]
```diff
@@ -18,25 +18,24 @@
  */
 
 "use strict";
-
-window.loadHistory = (function() {
+window.loadHistory = (() => {
```
Comment (The-Compiler) [inline qutebrowser/javascript/history.js]: Still not sure what to think about making every function a lambda - but I have no idea what's common here in the JS world.

**Comment by The-Compiler** [comment]
It also looks like your changes are conflicting in `scroll.js` with `master`, despite the last change to that being 11 days ago - are you on an outdated local branch?

**Comment by sandrosc** [comment]
Very nice, I love ES6!

---

Review status: 0 of 7 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[qutebrowser/javascript/.eslintrc.yaml, line 11 at r1](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/3217#-KxlMezoEK0FrmllJvoN:-KxlMezprRtH2JZz7-Aa:b-l8sj3b) ([raw file](https://github.com/qutebrowser/qutebrowser/blob/378498bbd7767a2277ab6fba9e54899959a7c972/qutebrowser/javascript/.eslintrc.yaml#L11)):*
> ```YAML
> 
> rules:
>      strict: ["error", "global"]
> ```

My personal preference is to enable strict mode, especially in es6 code.
I also believe it to be best practice. 

---

*[qutebrowser/javascript/history.js, line 21 at r1](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/3217#-KxjHWFECSBsPiPzlD7L-r1-21:-KxlPpFnry1pWk6QR6Lu:bd3xn7d) ([raw file](https://github.com/qutebrowser/qutebrowser/blob/186a30843a51bd65be988dc21285e8ea0f7a115c/qutebrowser/javascript/history.js#L21)):*
<details><summary><i>Previously, The-Compiler (Florian Bruhin) wrote…</i></summary><blockquote>

Still not sure what to think about making every function a lambda - but I have no idea what's common here in the JS world.
</blockquote></details>

I too would prefer a normal function.

---

*[qutebrowser/javascript/pac_utils.js, line 98 at r1](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/3217#-KxlT55PEpZ9CDzS0hKt:-KxlT55PEpZ9CDzS0hKu:b3stex8) ([raw file](https://github.com/qutebrowser/qutebrowser/blob/186a30843a51bd65be988dc21285e8ea0f7a115c/qutebrowser/javascript/pac_utils.js#L98)):*
> ```JavaScript
> }
> 
> const wdays = {
> ```

Why prefer quotes around object literal property names?
My preference would be to set https://eslint.org/docs/rules/quote-props to "as-needed".

---

*[qutebrowser/javascript/pac_utils.js, line 194 at r1](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/3217#-KxlXcAVj7I5aeL423Xz:-KxlXcAVj7I5aeL423Y-:b68it56) ([raw file](https://github.com/qutebrowser/qutebrowser/blob/378498bbd7767a2277ab6fba9e54899959a7c972/qutebrowser/javascript/pac_utils.js#L194)):*
> ```JavaScript
>     }
>     if (isGMT) {
>     var tmp = date;
> ```

Not sure if this can be deleted.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/3217#-:-KxlYpWvfFOZIs5mcHio:bhzbv5m)*

**Pull request closed without merging**